src/commit.c

```diff
@@ -890,12 +898,26 @@ int git_commit_create_with_signature(
 	const char *field;
 	const char *header_end;
 	git_buf commit = GIT_BUF_INIT;
+	git_commit *parsed;
+	git_array_oid_t parents = GIT_ARRAY_INIT;
 
-	/* We start by identifying the end of the commit header */
+	/* The first step is to verify that all the tree and parents exist */
+	parsed = git__calloc(1, sizeof(git_commit));
+	GIT_ERROR_CHECK_ALLOC(parsed);
+	if ((error = commit_parse(parsed, commit_content, strlen(commit_content), 0)) < 0)
+		goto cleanup;
+
+	if ((error = validate_tree_and_parents(&parents, repo, &parsed->tree_id, commit_parent_from_commit, parsed, NULL, true)) < 0)
+		goto cleanup;
```

Comment:
Is it certain that `parents` doesn't have to be cleared if `validate_tree_and_parents` returns an error?

Reply:
Looks like it to me - the error case of `validate_tree_and_parents` fn will clear the parents array before returning, if it had ever allocated anything into it.